.github/workflows/ci.yaml

```diff
@@ -0,0 +1,25 @@
+name: Component Graph CI
+
+on:
+  push:
+    branches: [ v0.x.x ]
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Print environment (debug)
+        run: env
+
+      - name: Fetch sources
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Run linter
+        run: cargo clippy
```

Comment:
I like being a bit more restrictive here:

```
cargo clippy -- -W clippy::unwrap_used -W clippy::expect_used -W clippy::panic
```

Or some variation of it. A fairly restrictive option that gives very good tips soemtimes:

```
cargo clippy -- -W clippy::restriction
```